airflow/dags/monitor/test_retrieval.py

```diff
@@ -0,0 +1,267 @@
+import json
+import logging
+import os
+from datetime import datetime
+from pathlib import Path
+from textwrap import dedent
+
+import pandas as pd
+from include.tasks.extract.utils.retrieval_tests import (
+    generate_answer,
+    get_or_create_drive_folder,
+    weaviate_search,
+    weaviate_search_mqr,
+)
+
+# from airflow.providers.weaviate.hooks.weaviate import WeaviateHook
+from include.tasks.extract.utils.weaviate.ask_astro_weaviate_hook import AskAstroWeaviateHook
+
+from airflow.decorators import dag, task
+from airflow.exceptions import AirflowException
+from airflow.models.param import Param
+from airflow.providers.google.suite.hooks.drive import GoogleDriveHook
+from airflow.providers.google.suite.hooks.sheets import GSheetsHook
+
+ask_astro_env = os.environ.get("ASK_ASTRO_ENV", "dev")
+test_questions_sheet_id = os.environ.get("TEST_QUESTIONS_SHEET_ID")
+askastro_endpoint_url = os.environ.get("ASK_ASTRO_ENDPOINT_URL")
+langchain_org_id = os.environ.get("LANGCHAIN_ORG")
+langchain_project_id = os.environ.get("LANGCHAIN_PROJECT_ID_" + ask_astro_env.upper(), None)
+azure_endpoint = os.environ["AZURE_OPENAI_USEAST_PARAMS"]
+google_domain_id = os.environ["GOOGLE_DOMAIN_ID"]
+
+drive_folder = "ask_astro/tests_" + ask_astro_env
+
+_WEAVIATE_CONN_ID = f"weaviate_{ask_astro_env}"
+_DRIVE_CONN_ID = "google_cloud_drive"
+
+WEAVIATE_CLASS = os.environ.get("WEAVIATE_CLASS", "DocsDev")
+
+logger = logging.getLogger("airflow.task")
+
+test_question_template_path = "include/data/test_questions_template.csv"
+
+expected_doc_count = 36860
+
+default_args = {"retries": 3, "retry_delay": 30, "trigger_rule": "none_failed"}
+
+
+@dag(
+    schedule_interval=None,
+    start_date=datetime(2023, 9, 27),
+    catchup=False,
+    is_paused_upon_creation=True,
+    default_args=default_args,
+    params={
+        "question_number_subset": Param(
+            default="",
+            title="Subset of test numbers to run.",
+            description=dedent(
+                """
+                A set of test ID numbers to run instead of the entire test set.
```

Comment:
I did not understand why we should not run the entire test set. Can you explain this please?

Reply:
This is optional.  At DAG run can select only a subset.  For instance, if there is a small change impacting only a couple of tests and you want to check just that subset without incurring the cost to run the entire suite.  Maybe need to run the subset tests a few times before confirming the fix.  After confirming that the fix works for that subset then run the whole suite to check for regression.